src/redux/products/products.sagas.js

```diff
@@ -47,6 +89,106 @@ export function* handleGetFilters() {
   }
 }
 
+export function* handleProductSpeciesLoad() {
+  try {
+    const categories = yield call(getProductCategories);
+    const species = yield call(getAllFilters);
+    yield put(setProductCategories(categories));
+    yield put(setAllFilterData(species));
+  } catch (e) {
+    yield call(handleProductsErrors, e);
+  }
+}
+
+export function* handleProductOptionsLoad() {
+  try {
+    const productOptions = yield call(getProductOptions);
+    yield put(setProductOptions(productOptions));
+  } catch (e) {
+    yield call(handleProductsErrors, e);
+  }
+}
+
+export function* handleModelsLoad({ payload }) {
+  try {
+    const models = yield call(getModelsByCategory, payload);
+    yield put(setModels(models));
+  } catch (e) {
+    yield call(handleProductsErrors, e);
+  }
+}
+
+export function* handleProductAdd() {
+  try {
+    yield put(setProductsLoading(true));
+    const productState = yield select(({ Products }) => Products);
+    const addedProduct = yield call(addProduct, productState);
```

Comment:
But we need also sort and filter our products after we added the new one, and sorting and filtering we have on the our backend, so that's the reason why i make new request